cot/src/test.rs

```diff
@@ -1578,3 +1586,360 @@ pub fn serial_guard() -> std::sync::MutexGuard<'static, ()> {
+    /// Returns an error if the cache could not be created.
+    ///
+    /// # Examples
+    ///
+    /// ```
+    /// use cot::test::TestCache;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() -> cot::Result<()> {
+    /// let test_cache = TestCache::new_memory();
+    /// let cache = test_cache.cache();
+    ///
+    /// // do something with the cache
+    ///
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[must_use]
+    pub fn new_memory() -> Self {
+        let cache = Cache::new(Memory::new(), None, Timeout::default());
+        Self::new(cache, CacheKind::Memory)
+    }
+
+    /// Create a new Redis test cache.
+    ///
+    /// The Redis URL is read from the `REDIS_URL` environment variable. If not
+    /// provided, it defaults to `redis://localhost`.
+    ///
+    /// Running with redis makes use of an internal allocator that selects what
+    /// DB a test will run. Every test requires its own database to avoid
+    /// conflicts. The allocator, by design, will reserve the last database
+    /// number for allocation purposes, so make sure your Redis instance is
+    /// configured with at least 2 databases. For example if your redis
+    /// instance has 16 logical databases, database 15 will be used for
+    /// allocations, and databases 0-14 will be used for tests.
+    ///
+    /// # Errors
+    ///
+    /// Returns an error if the Redis cache could not be created.
+    ///
+    /// # Panics
+    ///
+    /// Panics if Redis is not configured with at least 2 databases.
+    ///
+    /// # Examples
+    ///
+    /// ```no_run
+    /// use cot::test::TestCache;
+    ///
+    /// # #[tokio::main]
+    /// # async fn main() -> cot::Result<()> {
+    /// let test_cache = TestCache::new_redis().await?;
+    /// let cache = test_cache.cache();
+    ///
+    /// // do something with the cache
+    ///
+    /// # Ok(())
+    /// # }
+    /// ```
+    #[cfg(feature = "redis")]
+    pub async fn new_redis() -> Result<Self> {
+        let url = std::env::var("REDIS_URL").unwrap_or_else(|_| "redis://localhost".to_string());
+        let mut url = CacheUrl::from(url);
+
+        let redis = Redis::new(&url, 10)?;
+        let mut conn = redis.get_connection().await?;
+
+        let db_num = get_db_num(&mut conn).await;
+        assert!(
+            db_num > 1,
+            "Redis must be configured with at least 2 databases for testing"
+        );
+
+        let alloc_db = db_num - 1;
+
+        // switch to the allocation DB to perform initialization
+        set_current_db(&mut conn, db_num - 1).await;
+
+        let allocator = RedisDbAllocator::new(alloc_db, redis);
+        allocator.init().await?;
+        let current_db = allocator
+            .allocate()
+            .await?
+            .expect("Failed to allocate a Redis database for testing");
+
+        // create a new connection to the correct DB
+        url.inner_mut().set_path(current_db.to_string().as_str());
+        let redis = Redis::new(&url, 10)?;
```

Comment:
Let's move the `10` to a constant such as `DEFAULT_REDIS_POOL_SIZE` or something to avoid having this magic number here.